tools/ci-scripts/check-client-codegen-unit-tests

```diff
@@ -6,4 +6,5 @@
 
 set -eux
 cd smithy-rs
+printenv
```

Comment:
Weird, something in CI is setting `RUSTFLAGS=-D warnings`: https://github.com/smithy-lang/smithy-rs/actions/runs/6864720326/job/18667193959?pr=3192#step:3:356

Reply:
Oh, we bake these into the Docker image: https://github.com/smithy-lang/smithy-rs/blob/31d9821e0465218049c7345d14d23b97dd7bc011/tools/ci-build/Dockerfile#L194-L206

Reply:
I opened https://github.com/smithy-lang/smithy-rs/issues/3194.